### ISSUE TITLE
API objects can't make round trip

### ISSUE THREAD

**Issue by pneumee** [Azure/azure-sdk-for-python#523]
This may not be true for all objects retrieved via the various `get()` methods, but take attaching a data disk to a `VirtualMachine` instance for example. To attach a data disk to a VM, the process seems like it should be:

1. get the VM from the API
` vm = compute_client.virtual_machines.get(resource_group_name, vm_name)`

2. modify the local instance by appending a data disk to the VM
`vm.storage_profile.data_disks.append(data_disk)`

3. update the VM
`compute_client.virtual_machines.create_or_update(resource_group_name, vm)`

However, the following error is generated from API when attempting the above:

```
'{"error":{"code":"InvalidRequestContent","message":"The request content was invalid and could not be deserialized: \\'Could not find member \\'resources\\' on object of type \\'ResourceDefinition\\'. Path \\'resources\\', line 1, position 1507.\\'."}}'
```

This appears to be because the `extensions` property of the VM instance retrieved from the API generates an empty list, rather than a `None` value. When calling `create_or_update()`, the empty list starts to cause problems here:

https://github.com/Azure/azure-sdk-for-python/blob/master/azure-mgmt-compute/azure/mgmt/compute/computemanagement.py#L6697

Ultimately this is where the `resources` object is set (and is what the API is complaining about when the request is submitted):

https://github.com/Azure/azure-sdk-for-python/blob/master/azure-mgmt-compute/azure/mgmt/compute/computemanagement.py#L6805

So it seems like either a) the Python SDK  should treat empty lists (and maybe dictionaries?) the same as a `None` property value or b) the ARM API should handle this case more gracefully.

Obviously the workaround for this particular example is `vm.extensions = None` between the calls to `get()` and `create_or_update()`, but that seems like it shouldn't be necessary.

**Comment by lmazuel** [comment]
Hi,

I agree with this is a problem.
However at short term, we are currently finishing the 2.0.0 version of the SDK with a totally new code inside. I keep your bug open to check if the 2.0.0 is impacted by this problem.
Thank you,

**Comment by pneumee** [comment]
Thanks! Looking forward to 2.0.0

**Comment by lmazuel** [comment]
Hi,

Version 2.0.0a1 is just released. If you have the opportunity to test your issue in this new version, it would be awesome! 
Thank you!

**Comment by lmazuel** [comment]
This problem is fixed in the 2.0.0 version of the SDK. You can see an example in the official VM sample:
https://github.com/Azure-Samples/virtual-machines-python-manage/blob/master/example.py#L148

Thanks for the report! Feel free to open a new bug if you see something else.
